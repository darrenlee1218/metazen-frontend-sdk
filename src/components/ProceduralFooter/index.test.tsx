import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { ProceduralFooter } from '.';

describe('Procedural Footer', () => {
  it('should render a divider if requiredDivider is true', () => {
    const handleAdvance = jest.fn();
    const baseDom = render(<ProceduralFooter requireDivider onAdvance={handleAdvance} isAdvanceEnabled={false} />);
    expect(baseDom.container).toMatchSnapshot();
  });

  it('should render cancel button when onCancel is provided', () => {
    const handleCancel = jest.fn();
    const handleAdvance = jest.fn();
    const baseDom = render(
      <ProceduralFooter requireDivider onAdvance={handleAdvance} onCancel={handleCancel} isAdvanceEnabled={false} />,
    );
    expect(baseDom.container).toMatchSnapshot();
  });

  it('should not be able to call handleAdvance if isAdvanceEnabled is false', () => {
    const handleAdvance = jest.fn();
    const baseDom = render(<ProceduralFooter requireDivider onAdvance={handleAdvance} isAdvanceEnabled={false} />);

    fireEvent.click(baseDom.getByText('Next'));
    expect(handleAdvance).not.toHaveBeenCalled();
  });

  it('should be able to successfully call callbacks', () => {
    const handleCancel = jest.fn();
    const handleAdvance = jest.fn();
    const baseDom = render(
      <ProceduralFooter requireDivider isAdvanceEnabled onCancel={handleCancel} onAdvance={handleAdvance} />,
    );

    fireEvent.click(baseDom.getByText('Next'));
    expect(handleAdvance).toHaveBeenCalled();

    fireEvent.click(baseDom.getByText('Cancel'));
    expect(handleCancel).toHaveBeenCalled();
  });
});
