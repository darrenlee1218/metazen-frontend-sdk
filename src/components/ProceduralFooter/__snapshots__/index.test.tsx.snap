// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Procedural Footer should render a divider if requiredDivider is true 1`] = `
<div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-149q7or-MuiDivider-root"
  />
  <div
    class="MuiBox-root css-ea6zr6"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth Mui-disabled css-1aej8tm-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      disabled=""
      id="mui-1"
      tabindex="-1"
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Procedural Footer should render cancel button when onCancel is provided 1`] = `
<div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-149q7or-MuiDivider-root"
  />
  <div
    class="MuiBox-root css-ea6zr6"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-931uqb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Cancel
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth Mui-disabled css-1aej8tm-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      disabled=""
      id="mui-2"
      tabindex="-1"
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
